impl/maven-xml/src/main/java/org/apache/maven/internal/xml/DefaultXmlService.java

```diff
@@ -392,34 +392,6 @@ private static String getChildCombinationMode(Map<String, String> attributes) {
         return !isEmpty(value) ? value : DEFAULT_CHILDREN_COMBINATION_MODE;
     }
 
-    @Nullable
-    private static XmlNode findNodeById(@Nonnull List<XmlNode> nodes, @Nonnull String id) {
```

Comment:
one time usage so might be lost:

<img width="1906" alt="image" src="https://github.com/user-attachments/assets/93cf04f8-4ee2-4a49-bdc0-4e7486253e60" />